### DISCUSSION TITLE
GSOC 2022 : Interested in project "Improving shadowOf generation".

### DISCUSSION THREAD

**Discussion by Akshay2131** [robolectric/robolectric#7181] [General]
Hello guys,
This side Akshay Sharma and i am currently pursuing bachelor's in computer sciencee and engineering at National Institute of Technology Hamirpur. I have studied computer science subjects like operating systems, data structures, OOPs and many more. I am well versed with java , c++ and working with android development nowadays.
I want to work with this project. So, can you tell me more about this and guide me further about the same.

**Comment by hoisie** [comment]
@Akshay2131 as a first step it would be great to set up a development environment where you can build Robolectric. Could you try to follow the instructions at http://robolectric.org/building-robolectric/ and see if you can get to the point where you can build Robolectric? What OS are you using?

**Reply by Akshay2131** [reply]
All right.. I am working to set up environment soon.. I am using ubuntu as OS

**Comment by Akshay2131** [comment]
@hoisie In Shadows.java file, it shows Cannot resolve symbol 'Shadows'.
Do you have any idea/suggestion about it?

**Reply by Akshay2131** [reply]
Actually, i got it that why it was there..

**Comment by Akshay2131** [comment]
@hoisie  i have created a new PR as Add getCurrentSyncs() in ShadowContentResolver #7225 
Could you please review it?

**Reply by utzcoz** [reply]
You can ping at your PR page directly.

**Comment by Akshay2131** [comment]
@hoisie and @utzcoz  could you guys review it and suggest something if needed to be added in this GSoC proposal..
https://docs.google.com/document/d/1UlX41UOxmhHZ0R4DL1djRLp-3WBz8N9DktQyn5lNa9c/edit?usp=sharing

**Reply by utzcoz** [reply]
I have left some comments. Almost of it LGTM.

**Reply by Akshay2131** [reply]
Thanks @utzcoz to review it..

**Reply by hellosagar** [reply]
@Akshay2131 your proposal looks sick!
Mentors, is that `Project Timeline` section is mandatory to have in the proposal because I've submitted it without that, other than that I covered all the sections

**Reply by utzcoz** [reply]
IMHO, it's not mandatory. And @hellosagar please be friendly.

**Reply by utzcoz** [reply]
There is an official tutorial about proposal: https://google.github.io/gsocguides/student/writing-a-proposal

**Reply by hellosagar** [reply]
> IMHO, it's not mandatory. And @hellosagar please be friendly.

ohh sorry for the confusion, I just mean it looks cool/good/impressive

**Reply by hellosagar** [reply]
sorry for the confusion I'm not a native English speaker but I just meant that It is well thought out

![Screenshot 2022-04-19 at 12 08 32 PM](https://user-images.githubusercontent.com/50016799/163941234-0549fef6-6573-4101-ad10-062896755e91.png)

**Reply by utzcoz** [reply]
I'm not native English speaker too. Sorry for misunderstanding it.

**Reply by hellosagar** [reply]
The bottom line is that I was just giving a compliment, I will try to use neutral words only from now on and will avoid text language and be more professional

**Comment by hoisie** [comment]
@Akshay2131 I have made some comments on the doc. The main part of the project will involve updating [ImplementsValidator](https://github.com/robolectric/robolectric/blob/master/processor/src/main/java/org/robolectric/annotation/processing/validator/ImplementsValidator.java) to keep track of which API levels for all shadow classes, and make sure that shadowOf generation are disabled for classes that are too new. Another part of the project is going through shadows that have `isInAndroidSdk = false` and potentially removing that. If you make those changes I can take another pass tomorrow.

**Reply by Akshay2131** [reply]
Got it! I will make the changes suggested by you shortly..
Thanks@hoisie

**Comment by Akshay2131** [comment]
I have marked the changes in the proposal as suggested by @hoisie..

**Comment by Akshay2131** [comment]
![Screenshot from 2022-04-19 11-13-16](https://user-images.githubusercontent.com/54117867/164059959-3c1feb5a-e5d6-457c-81b8-d02b7dd36d05.png)

**Reply by Akshay2131** [reply]
I have submitted the proposal. Thanks @hoisie and @utzcoz  for their help..

**Comment by Akshay2131** [comment]
I just checked that I got selected for GSoC 2022.
Thanks to @hoisie and @utzcoz for their constant support and belief in me to select me as a GSoC contributor this year.
Hoping to enjoy this journey with u guys..

**Comment by hoisie** [comment]
Can you paste the exact error message? There may be another compiler error
that is the real cause. Shadows.java gets generated by a Java annotation
professor, so if there is another compile error, it won't be generated.

**Comment by utzcoz** [comment]
GSoC 2022 was finished, and related discussions should be posted on related issues or other discussion page.
